lib/exabgp/bgp/message/update/attribute/bgpls/linkstate.py

```diff
@@ -72,11 +75,52 @@ def __init__ (self, code, rep):
 		self.code = code
 
 	def __repr__ (self):
-		return "Unknown attribute is: %s" % (self.rep)
+		return "Attribute with code [ %s ] not implemented" % (self.code)
 
 	@classmethod
 	def unpack (cls,scode,data):
 		length = len(data)
 		info = binascii.b2a_uu(data[:length])
 		return cls(code=scode,rep=info)
 
+	def json (self,compact=None):
+		return '"attribute-not-implemented": "%s"' % (self.code)
+
+
+class LsGenericFlags(object):
+
+#	draft-ietf-isis-segment-routing-extensions Prefix-SID Sub-TLV
+	ISIS_SR_FLAGS = ['R', 'N', 'P', 'E', 'V', 'L', 'RSV', 'RSV']
+#	RFC 7794 IPv4/IPv6 Extended Reachability Attribute Flags
+	ISIS_SR_ATTR_FLAGS = ['X', 'R', 'N', 'RSV', 'RSV', 'RSV', 'RSV', 'RSV']
+#	draft-ietf-isis-segment-routing-extensions - Adj-SID IS-IS Flags
+	ISIS_SR_ADJ_FLAGS = ['F', 'B', 'V', 'L', 'S', 'P', 'RSV', 'RSV']
+# 	isis-segment-routing-extensions 3.1. SR-Capabilities Sub-TLV
+	ISIS_SR_CAP_FLAGS = ['I', 'V', 'RSV', 'RSV', 'RSV', 'RSV', 'RSV', 'RSV']
+#	RFC 7752 3.3.1.1. Node Flag Bits TLV
+	LS_NODE_FLAGS = ['O', 'T', 'E', 'B', 'R', 'V', 'RSV', 'RSV']
+#	RFC 7752 3.3.3.1. IGP Flags TLV
+	LS_IGP_FLAGS = ['D', 'N', 'L', 'P', 'RSV', 'RSV', 'RSV', 'RSV']
+#	RFC 7752 3.3.2.2.  MPLS Protocol Mask TLV
+	LS_MPLS_MASK = ['LDP', 'RSVP-TE', 'RSV', 'RSV', 'RSV', 'RSV', 'RSV', 'RSV']
+#	RFC 5307 1.2.
+	LS_PROTECTION_MASK = ['ExtraTrafic', 'Unprotected', 'Shared', 'Dedicated 1:1',
+							'Dedicated 1+1', 'Enhanced', 'RSV', 'RSV']
+
+	def __init__(self,flags):
+		self.flags = flags
+
+	@classmethod
+	def unpack(cls,data,pattern):
+		pad = pattern.count('RSV')
+		repeat = len(pattern) - pad
+		flag_array = binascii.b2a_hex(data)
+		hex_rep = hex(int(flag_array, 16))
+		bit_array = BitArray(hex_rep)
+		valid_flags = [''.join(item) + '0' * pad for item in itertools.product('01', repeat=repeat)]
```

Comment:
This is not python 3 friendly .. I think we have a function to perform string concatenation